session/grpc.go

```diff
@@ -79,21 +81,54 @@ func monitorHealth(ctx context.Context, cc *grpc.ClientConn, cancelConn func())
 	defer cancelConn()
 	defer cc.Close()
 
-	ticker := time.NewTicker(1 * time.Second)
+	ticker := time.NewTicker(5 * time.Second)
 	defer ticker.Stop()
 	healthClient := grpc_health_v1.NewHealthClient(cc)
 
+	failedBefore := false
+	consecutiveSucessful := 0
+	maxHealthcheckDuration := 30 * time.Second
```

Comment:
This name feels misleading to me because the healthcheck timeout can exceed 30 seconds. Maybe `defaultHealthcheckTimeout`?